src/printer.rs

```diff
@@ -31,32 +32,48 @@ impl<'a> Printer<'a> {
     }
 
     pub fn print_header(&mut self, filename: Option<&str>) -> Result<()> {
-        match self.options.style {
-            OptionsStyle::Full => {}
-            _ => return Ok(()),
+        if !self.options
+            .output_components
+            .contains(&OutputComponent::Header)
+        {
+            return Ok(());
         }
 
-        self.print_horizontal_line('┬')?;
-
-        write!(
-            self.handle,
-            "{}{} ",
-            " ".repeat(PANEL_WIDTH),
-            self.colors.grid.paint("│"),
-        )?;
-
-        writeln!(
-            self.handle,
-            "{}{}",
-            filename.map_or("", |_| "File: "),
-            self.colors.filename.paint(filename.unwrap_or("STDIN"))
-        )?;
-
-        self.print_horizontal_line('┼')
+        if self.options
+            .output_components
+            .contains(&OutputComponent::Grid)
+        {
+            self.print_horizontal_line('┬')?;
```

Comment:
Since this function is only relevant for grid, you could put a check in the function itself to print nothing if grid is not present. it would simplify a lot. what do you think?

Reply:
That depends: the way it is implemented here right now, the header can also be shown regardless of if the grid is enabled or not. I suppose this is a decision to make: do we want to display a header if the grid is not requested and if so, how should it be formatted?

Reply:
Sorry if I wasn't clear, I was actually specifically talking about `print_horizontal_line`, but you're right about the header thing. 🤷‍♂️